datacube/storage/_rio.py

```diff
@@ -31,19 +34,27 @@ def _rasterio_crs_wkt(src):
         return ''
 
 
+def maybe_lock(lock):
+    if lock is None:
+        return contextlib.suppress()
```

Comment:
Is this meant to be a no-op?

Reply:
yes this noop context manager, it is usually used to suppress a set of exceptions, but without any arguments it's just a thing you can call `with ` with, except you can't call `with as`.